src/detectors.rs

```diff
@@ -40,6 +40,17 @@ impl ThreadInfo {
             pthread_id: get_thread_id(),
         }
     }
+
+    // Getter for the id field
+    pub fn get_id(&self) -> &ThreadId {
+        &self.id
+    }
+
+    // Getter for the pthread_id field (only available on Unix)
+    #[cfg(unix)]
+    pub fn get_pthread_id(&self) -> &libc::pthread_t {
```

Comment:
In general, the Rust convention for accessors does not include the word `get_` in method names (see: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).

I think we should change these to:
```suggestion
    /// Returns the thread's [`ThreadId`].
    pub fn id(&self) -> &ThreadId {
        &self.id
    }

    /// Returns the [`libc::pthread_t`] ID of this thread (only available on Unix).
    #[cfg(unix)]
    pub fn pthread_id(&self) -> &libc::pthread_t {
```